File: domain/src/main/java/com/crhonvas/domain/repo/ICommentsRepository.java
package com.crhonvas.domain.repo;


import com.crhonvas.domain.model.comments.Comment;

import java.util.List;

import io.reactivex.Single;


public interface ICommentsRepository {
    Single<Comment> getCommentById(Integer commentId);

    Single<List<Comment>> getComments();

    Single<List<Comment>> getCommentsForPostId(Integer commentId);
}

File: domain/src/main/java/com/crhonvas/domain/repo/ICommentsRepository.kt
package com.crhonvas.domain.repo

import com.crhonvas.domain.model.comments.Comment
import io.reactivex.Single

interface ICommentsRepository {
    fun getCommentById(commentId: Integer): Single<Comment>
    fun getComments(): Single<List<Comment>>
    fun getCommentsForPostId(postId: Integer): Single<List<Comment>>
}